skrifa/src/outline/base.rs

```diff
@@ -0,0 +1,58 @@
+//! Common functionality for glyf, cff and autohinting scalers.
+
+use raw::{
+    tables::{hmtx::Hmtx, hvar::Hvar},
+    types::{BigEndian, F2Dot14, GlyphId, Tag},
+    FontRef, TableProvider,
+};
+
+/// Common functionality for glyf, cff and autohinting scalers.
+#[derive(Clone)]
+pub(crate) struct BaseOutlines<'a> {
```

Comment:
When I see cff or glyf structs holding base: BaseOutlines I'm left unclear what that does. I definitely wouldn't have anticipated the FontRef hides here.

Base is strongly associated with something you derive from for many of us and doesn't really conveying the common bits concept. 

To ask the obvious, why not write a trait and implement it for FontRef? - cff and glyf holding `font: FontRef` and calling e.g. `font.lsb` would read more clearly to me. The main service this seems to serve is to avoid calling `font.hmtx` but if that's too slow ... maybe we could make it not be?

Reply:
Perf measurements suggest that looking up the tables on demand roughly doubles the cost of metrics retrieval. This may be acceptable for static text, but the animated variable case is one we like to highlight so wasting cycles for the sake of purity seems ill advised.

I renamed base -> common and BaseOutlines to OutlinesCommon for now. I agree this isn't ideal but I'd like to make progress on autohinting so I filed #1085 to revisit this in the future.